### PR TITLE
feat: add checks for header case and last character

### PR DESCRIPTION
**Pull request by dzhu**
I came across this project and loved the overall idea and the imperative check, but I noticed that some other common guidelines (restrictions on the leading case and trailing punctuation of the header) were missing; this adds checks to enforce those. The desired case can be specified as either "upper" or "lower"; the punctuation check is implemented as a general check for disallowing the last character in the header to be from any given set of characters.

### CODE REVIEW

**Comment by andrewrynhard** [comment]
@dzhu Love the changes. Looks like tests are failing though.

File: internal/policy/commit/commit.go [hunk 1]
```diff
@@ -33,6 +33,10 @@ type Commit struct {
 	RequireCommitBody bool `mapstructure:"requireCommitBody"`
 	// Conventional is the user specified settings for conventional commits.
 	Conventional *Conventional `mapstructure:"conventional"`
+	// HeaderCase is the case that the first word of the header must have ("upper" or "lower").
+	HeaderCase string `mapstructure:"headerCase"`
+	// HeaderInvalidLastCharacters is a string containing all invalid last characters for the header.
+	HeaderInvalidLastCharacters string `mapstructure:"headerInvalidLastCharacters"`
```
Comment (andrewrynhard) [inline internal/policy/commit/commit.go]: Should we have a set of default characters?

Comment (andrewrynhard) [inline internal/policy/commit/commit.go]: What do you think of something like:

```
header:
  case: ...
  headerInvalidLastCharacters:
```

to keep all the header related settings together.

Comment (andrewrynhard) [reply internal/policy/commit/commit.go]: We could have `body` and `footer` settings later on, and they could follow the same design.

Comment (dzhu) [reply internal/policy/commit/commit.go]: That makes a lot of sense; I'll work on it! Should the current `headerLength` and `imperative` also be moved there?

Comment (dzhu) [reply internal/policy/commit/commit.go]: Hmm, how would that show up in config? I played around with the `mapstructure` definitions and I didn't see a good way to expose a default.

**Comment by dzhu** [comment]
I've updated the config structure, including moving the length and imperative checks inside the new subfield. I also went ahead and updated `README.md` and `.conform.yaml` to put in the new checks, since I thought that made sense, but I'm happy to put any of that back if you'd like.

**Comment by andrewrynhard** [comment]
@dzhu Thanks! I like all the header options under one struct much better. Since this is a backwards incompatible change, I'll need some time to get things in order before I cut a release that includes this change. Appreciate the time you spent on this!

**Comment by dzhu** [comment]
Sounds good, thanks! I had fun working on this.

Roughly speaking, will the time to the release be in the range of days or weeks? I obviously don't want to rush you at all, but I'd like to use the new features at my workplace, and it would be helpful to know for planning. (Looks like we can `go get` a specific commit, so we're not even totally blocked on a release, but the rest of the team might like it a bit better if there is one.)

**Comment by andrewrynhard** [comment]
@dzhu looking more like a week. Caught up in another project right now and it’s release week 🙂

**Pull request merged**